src/utils/v2/isArray.ts

```diff
@@ -0,0 +1,8 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export function isArray<T>(maybeArray: T[] | null | undefined): maybeArray is T[] {
```

Comment:
Is this really necessary? Doesn't it just replace a one-liner with a different one-liner?

Reply:
Strictly necessary, no.  However, the utility is useful as a typeguard in mapping functions (e.g. `collection.filter(isArray)`); `Array.isArray()` doesn't indicate type, so you're then forced to cast after the check.